### PR TITLE
[protocol] New request `coq/getDocument` to get serialized document.

### PR DESCRIPTION
**Pull request by ejgallego**
cc: https://github.com/ejgallego/coq-serapi/pull/318

### CODE REVIEW

**Comment by ejgallego** [comment]
Tested, seems to work well; let's wait for some more feedback from Alectryon devs.

File: editor/code/package.json [hunk 1]
```diff
@@ -88,6 +88,10 @@
       {
         "command": "coq-lsp.goals",
         "title": "Coq LSP: Show Goals at Point"
+      },
+      {
+        "command": "coq-lsp.document",
+        "title": "Coq LSP: Get document in serialized format, print in console"
```
Comment (Alizter) [inline editor/code/package.json:94]: Do we want to expose this in the client?

**Comment by ejgallego** [comment]
@cpitclaudel this seems to work fine in my testing, modulo a JSON serialization todo in `serlib` upstream, what do you think?

File: editor/code/package.json (hunk 1 above)
Comment (ejgallego) [reply editor/code/package.json:94]: Not sure, maybe we could indeed safe the stuff in a file. For now I need it for testing easily.

**Comment by cpitclaudel** [comment]
Looks good! Thanks a lot :)

**Comment by ejgallego** [comment]
Ok thanks, will add to `main` for now, I guess more API will have to be exposed, but for now this should allow you to start, and also includes completion status.

The current model of the server is that it will only run the request when the full document has checked, but a position could be included, which would mean "check to this point, run `getDocument`"

**Pull request merged**